Update for Webpack 5, emit string assets only once

silverwind opened the pull request:
Commit 1 adds Webpack 5 compatibilty, fixes https://github.com/sindresorhus/add-asset-webpack-plugin/issues/5.
Commit 2 prevents string assets to be emitted multiple times, fixes https://github.com/sindresorhus/add-asset-webpack-plugin/issues/4.

silverwind committed:
Update for webpack 5

Update the plugin to webpack 5's API. Theoretically, it should work on
webpack >= 4.40.0 but I think it's better to require 5 because webpack
4 users can just stay on version 1.0.0 of this plugin.

Fixes: https://github.com/sindresorhus/add-asset-webpack-plugin/issues/5

silverwind commented on fixture/webpack.config.js:
```diff
@@ -5,6 +5,9 @@ module.exports = {
 	},
+	optimization: {
+		minimize: false
+	},
```
In case you're wondering why this is disabled: Webpack 5 by default runs terser on JS assets but it would throw up because the JS files created during the test are invalid JS (containing only a single emoji).

silverwind replied:
Actually, I fixed this by emitting `.txt` which terser will not act upon.

silverwind replied:
Was changed again to emit valid JS files in the test.

silverwind commented:
ping @sindresorhus

sibiraj-s commented on index.js:
```diff
@@ -1,19 +1,23 @@
 'use strict';
 
+const {Compilation} = require('webpack');
+const {RawSource} = require('webpack-sources');
```
Remove this dependency. webpack 5 already exports it.

Use

```js
const {Compilation, sources} = require('webpack');
const {RawSource} = sources
```

sibiraj-s commented on package.json:
```diff
@@ -28,14 +28,17 @@
 		"file"
 	],
+	"dependencies": {
+		"webpack-sources": "^2.2.0"
```
Remove this.

silverwind committed:
Emit string assets only once

Fixes: https://github.com/sindresorhus/add-asset-webpack-plugin/issues/4

silverwind commented:
Fixed the import to be directly from webpack so we can omit the `webpack-sources` dependency.

Added a second commit which skips emitting string assets if they already exist in the compilation.

sibiraj-s reviewed (approved):
Looks good.
